raiden/connection_manager.py

```diff
@@ -0,0 +1,265 @@
+        """Connect to the network.
+        Use this to establish a connection with the token network.
+        Args:
+            funds (int): the amount of tokens spendable for this
+                         ConnectionManager.
+        """
+        if funds <= 0:
+            raise ValueError('connecting needs a positive value for `funds`')
+        with self.lock:
+            self.funds = funds
+            funding = self.initial_funding_per_partner
+            for partner in self.find_new_partners(self.initial_channel_target):
+                self.raiden.api.open(
+                    self.token_address,
+                    partner,
+                )
+                self.raiden.api.deposit(
+                    self.token_address,
+                    partner,
+                    funding
+                )
+
+    def leave(self, wait_for_settle=True, max_wait=30):
+        """
+        Leave the token network.
+        This implies closing all open channels and optionally wait for
+        settlement.
+        Args:
+            wait_for_settle (bool): block until successful settlement?
+            max_wait (float): maximum time to wait
+        """
+        with self.lock:
+            self.initial_channel_target = 0
+            open_channels = self.open_channels
+            channel_specs = [(
+                self.token_address,
+                c.partner_address) for c in open_channels]
+            for channel in channel_specs:
+                try:
+                    self.raiden.api.close(*channel),
+                except RuntimeError:
+                    # if the error wasn't that the channel was already closed: raise
+                    if channel[1] in [c.partner_address for c in self.open_channels]:
+                        raise
+
+            # force state update
+            self.raiden.poll_blockchain_events(self.raiden.get_block_number())
+
+            if wait_for_settle:
+                timeout = time.time() + max_wait
+                while any(c.state != CHANNEL_STATE_SETTLED for c in open_channels):
+                    # force state update
+                    self.raiden.poll_blockchain_events(self.raiden.get_block_number())
+                    if time.time() > timeout:
+                        log.debug(
+                            'timeout while waiting for settlement',
+                            unsettled=sum(
+                                1 for channel in open_channels if
+                                channel.state != CHANNEL_STATE_SETTLED
+                            ),
+                            settled=sum(
+                                1 for channel in open_channels if
+                                channel.state == CHANNEL_STATE_SETTLED
+                            )
+                        )
+                        break
+
+    def join_channel(self, partner_address, partner_deposit):
+        """Will be called, when we were selected as channel partner by another
```

Comment:
Can you please consistently either do `"""` and then new line or `"""Start of comment`. I prefer the former.